src/xdocs/config_coding.xml

```diff
@@ -4897,6 +4897,84 @@ if (&quot;something&quot;.equals(x))
       </subsection>
     </section>
 
+    <section name="UnnecessarySemicolonInEnumeration">
+      <p>Since Checkstyle 8.22</p>
+      <subsection name="Description" id="UnnecessarySemicolonInEnumeration_Description">
+        <p>
+            Checks if unnecessary semicolon is in enum definitions.
+            Semicolon is not needed if enum body is empty.
+            The check will flag the following with warnings:
+        </p>
+        <source>
```

Comment:
Please move example of violations and good code to `Examples` section.

Reply:
That can be done.
I took as an example UnnecessaryParentheses, there is a code example just below description, so I made same thing. But I see that some other checks have examples in separate section.

Imo there should be some policy to clarify where to put examples.

Reply:
May be somewhere on contribution page. Brief description with relevant examples for existing checks would be great to make it easier to understand relevant requirements

Reply:
Unfortunately I do not know the perfect example of a check with a well written documentation.

I moved this discussion to https://github.com/checkstyle/checkstyle/issues/6789

Reply:
Probably https://checkstyle.org/config_annotation.html#AnnotationOnSameLine is good example

https://checkstyle.org/config_annotation.html#AnnotationLocation is also fine, but some minimum of code samples is in description

Reply:
Updated, please take a look